[jvm-package] looser throw specifier for ‘virtual xgboost::data::SimpleDMatrix::~SimpleDMatrix() noexcept (false)’

XGBoost base version: 0.72
Git cherry-pick: https://github.com/dmlc/xgboost/pull/3697/commits/d95c2b2591eaa773df44bb9ecf03561e45bdc53b
Dev environment: Red Hat Enterprise Linux Workstation 7.3

Cloned `dmlc-core` and `rabbit` to the `xgboost` repository.
Run `./build.sh`.

Exception trace:
In file included from src/data/data.cc:10:0:
src/data/./simple_dmatrix.h:20:7: error: looser throw specifier for ‘virtual xgboost::data::SimpleDMatrix::~SimpleDMatrix() noexcept (false)’
 class SimpleDMatrix : public DMatrix {
       ^
In file included from src/data/data.cc:5:0:
include/xgboost/data.h:297:11: error:   overriding ‘virtual xgboost::DMatrix::~DMatrix() noexcept (true)’
   virtual ~DMatrix() = default;
           ^
make: *** [build/data/data.o] Error 1

Should I include some additional commits to make the cherry-picked one work?

Run mvn package -DskipTests?

On Fri, Jun 14, 2019 at 4:35 PM Chenya Zhang <notifications@github.com>
wrote:

> XGBoost base version: 0.72
> Git cherry-pick: d95c2b2
> <https://github.com/dmlc/xgboost/commit/d95c2b2591eaa773df44bb9ecf03561e45bdc53b>
> Dev environment: Red Hat Enterprise Linux Workstation 7.3
>
> Cloned dmlc-core and rabbit to the xgboost repository.
> Run ./build.sh.
>
> Exception trace:
> In file included from src/data/data.cc:10:0:
> src/data/./simple_dmatrix.h:20:7: error: looser throw specifier for
> ‘virtual xgboost::data::SimpleDMatrix::~SimpleDMatrix() noexcept (false)’
> class SimpleDMatrix : public DMatrix {
> ^
> In file included from src/data/data.cc:5:0:
> include/xgboost/data.h:297:11: error: overriding ‘virtual
> xgboost::DMatrix::~DMatrix() noexcept (true)’
> virtual ~DMatrix() = default;
> ^
> make: *** [build/data/data.o] Error 1
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dmlc/xgboost/issues/4567?email_source=notifications&email_token=AAFFQ6GH2XVXVL7Z4TIECQLP2QTKXA5CNFSM4HYNXRL2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4GZVHM4A>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFFQ6A7K4ZIMWCWUNXZ6FLP2QTKXANCNFSM4HYNXRLQ>
> .
>

--- a/src/data/sparse_page_dmatrix.h
+++ b/src/data/sparse_page_dmatrix.h
@@ -24,6 +24,7 @@ class SparsePageDMatrix : public DMatrix {
   explicit SparsePageDMatrix(std::unique_ptr<DataSource>&& source,
                              std::string cache_info)
       : row_source_(std::move(source)), cache_info_(std::move(cache_info)) {}
+  virtual ~SparsePageDMatrix() = default;
 
   MetaInfo& Info() override;
 

